src/physics.cpp

```diff
@@ -0,0 +1,1174 @@
+}
+
+void
+create_fission_sites(Particle* p, int i_nuclide, const Reaction* rx, Bank* bank_array,
+  int64_t* size_bank, int64_t bank_capacity)
+{
+  // TODO: Heat generation from fission
+
+  // If uniform fission source weighting is turned on, we increase or decrease
+  // the expected number of fission sites produced
+  double weight = settings::ufs_on ? ufs_get_weight(p) : 1.0;
+
+  // Determine the expected number of neutrons produced
+  double nu_t = p->wgt / simulation::keff * weight * simulation::micro_xs[
+    i_nuclide-1].nu_fission / simulation::micro_xs[i_nuclide-1].total;
+
+  // Sample the number of neutrons produced
+  int nu = static_cast<int>(nu_t);
+  if (prn() <= (nu_t - nu)) ++nu;
+
+  // Check for the bank size getting hit. For fixed source calculations, this
+  // is a fatal error; for eigenvalue calculations, it just means that k-eff
+  // was too high for a single batch.
+  if (*size_bank + nu > bank_capacity) {
+    if (settings::run_mode == RUN_MODE_FIXEDSOURCE) {
+      throw std::runtime_error{"Secondary particle bank size limit reached."
+           " If you are running a subcritical multiplication problem,"
+           " k-effective may be too close to one."};
+    } else {
+      if (mpi::master) {
+        warning("Maximum number of sites in fission bank reached. This can"
+          " result in irreproducible results using different numbers of"
+          " processes/threads.");
+      }
+    }
+  }
+
+  // Begin banking the source neutrons
+  // First, if our bank is full then don't continue
+  if (nu == 0 || *size_bank == bank_capacity) return;
+
+  // Initialize the counter of delayed neutrons encountered for each delayed
+  // group.
+  double nu_d[MAX_DELAYED_GROUPS] = {0.};
+
+  p->fission = true;
+  for (size_t i = *size_bank; i < std::min(*size_bank + nu, bank_capacity); ++i) {
+    // Bank source neutrons by copying the particle data
+    bank_array[i].xyz[0] = p->coord[0].xyz[0];
+    bank_array[i].xyz[1] = p->coord[0].xyz[1];
+    bank_array[i].xyz[2] = p->coord[0].xyz[2];
+
+    // Set that the bank particle is a neutron
+    bank_array[i].particle = static_cast<int>(ParticleType::neutron);
+
+    // Set the weight of the fission bank site
+    bank_array[i].wgt = 1. / weight;
+
+    // Sample delayed group and angle/energy for fission reaction
+    sample_fission_neutron(i_nuclide, rx, p->E, &bank_array[i]);
+
+    // Set the delayed group on the particle as well
+    p->delayed_group = bank_array[i].delayed_group;
+
+    // Increment the number of neutrons born delayed
+    if (p->delayed_group > 0) {
+      nu_d[p->delayed_group-1]++;
+    }
+  }
+
+  // Increment number of bank sites
+  *size_bank = std::min(*size_bank + nu, bank_capacity);
+
+  // Store the total weight banked for analog fission tallies
+  p->n_bank = nu;
+  p->wgt_bank = nu / weight;
+  for (size_t d = 0; d < MAX_DELAYED_GROUPS; d++) {
+    p->n_delayed_bank[d] = nu_d[d];
+  }
+}
+
+// void sample_photon_reaction(Particle* p)
```

Comment:
I moved the entire source of physics.F90 to physics.cpp so that I could do wholesale search-and-replace for things like `end if` -> `}`. But, some of the photon routines will require more data from the h5 files, so again to limit the scope of this PR, I left these commented out for now. I'm planning on finishing the conversion of photon routines in my next PR, but will put a note here for now.